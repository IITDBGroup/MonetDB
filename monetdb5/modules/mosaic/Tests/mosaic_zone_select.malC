# Compression of noncompressable column


b:= bat.new(:int);
bat.append(b,50);
bat.append(b,19531015);
bat.append(b,9);
b:= bat.append(b,b);
base:= algebra.copy(b);

io.print(base);
mosaic.compress(b,"zone");
s:= algebra.select(base,0,50,false,false,false);
io.print(s);
xs:= mosaic.select(b,0,50,false,false,false);
io.print(xs);

s:= algebra.select(base,0,50,false,true,false);
io.print(s);
xs:= mosaic.select(b,0,50,false,true,false);
io.print(xs);

s:= algebra.select(base,0,50,true,false,false);
io.print(s);
xs:= mosaic.select(b,0,50,true,false,false);
io.print(xs);

s:= algebra.select(base,0,50,true,true,false);
io.print(s);
xs:= mosaic.select(b,0,50,true,true,false);
io.print(xs);

s:= algebra.select(base,nil:int,50,false,false,false);
io.print(s);
xs:= mosaic.select(b,nil:int,50,false,false,false);
io.print(xs);

s:= algebra.select(base,50,nil:int,false,false,false);
io.print(s);
xs:= mosaic.select(b,50,nil:int,false,false,false);
io.print(xs);

s:= algebra.select(base,nil:int,nil:int,false,false,false);
io.print(s);
xs:= mosaic.select(b,nil:int,nil:int,false,false,false);
io.print(xs);

s:= algebra.select(base,nil:int,nil:int,false,false,true);
io.print(s);
xs:= mosaic.select(b,nil:int,nil:int,false,false,true);
io.print(xs);

# using candidatelists
io.print("candidate stuff");
c:= bat.new(:oid);
bat.append(c,2@0);
bat.append(c,5@0);

s:= algebra.select(base,c,0,50,false,false,false);
io.print(s);
xs:= mosaic.select(b,c,0,50,false,false,false);
io.print(xs);

s:= algebra.select(base,c,0,50,false,true,false);
io.print(s);
xs:= mosaic.select(b,c,0,50,false,true,false);
io.print(xs);

s:= algebra.select(base,c,0,50,true,false,false);
io.print(s);
xs:= mosaic.select(b,c,0,50,true,false,false);
io.print(xs);

s:= algebra.select(base,c,0,50,true,true,false);
io.print(s);
xs:= mosaic.select(b,c,0,50,true,true,false);
io.print(xs);

s:= algebra.select(base,c,nil:int,50,false,false,false);
io.print(s);
xs:= mosaic.select(b,c,nil:int,50,false,false,false);
io.print(xs);

s:= algebra.select(base,c,50,nil:int,false,false,false);
io.print(s);
xs:= mosaic.select(b,c,50,nil:int,false,false,false);
io.print(xs);

s:= algebra.select(base,c,nil:int,nil:int,false,false,false);
io.print(s);
xs:= mosaic.select(b,c,nil:int,nil:int,false,false,false);
io.print(xs);

s:= algebra.select(base,c,nil:int,nil:int,false,false,true);
io.print(s);
xs:= mosaic.select(b,c,nil:int,nil:int,false,false,true);
io.print(xs);

