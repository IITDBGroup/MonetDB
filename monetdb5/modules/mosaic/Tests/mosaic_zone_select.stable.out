stdout of test 'mosaic_zone_select` in directory 'monetdb5/modules/mosaic` itself:


# 17:24:14 >  
# 17:24:14 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=33832" "--set" "mapi_usock=/var/tmp/mtest-18516/.s.monetdb.33832" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/export/scratch1/mk/mosaic//Linux/var/MonetDB/mTests_monetdb5_modules_mal" "mosaic_zone_subselect.mal"
# 17:24:14 >  

# MonetDB 5 server v11.20.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_modules_mal', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 15.591 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2014 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.ins.cwi.nl:33832/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-18516/.s.monetdb.33832
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/R   module loaded

# 11:50:50 >  
# 11:50:50 >  "mclient" "-lmal" "-ftest" "-tnone" "-Eutf-8" "--host=/var/tmp/mtest-30332" "--port=39586"
# 11:50:50 >  

# 15:01:31 >  
# 15:01:31 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-15809" "--port=37660"
# 15:01:31 >  

#io.print(base);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	50  ]
[ 1@0,	19531015  ]
[ 2@0,	9  ]
[ 3@0,	50  ]
[ 4@0,	19531015  ]
[ 5@0,	9  ]
#header block 21c10b0 version 20140808
#index top 1
#[0] 6 112
#blk  type 5 todo 0
#wins 1 0 0 0 0 0 
#elms 6 0 0 0 0 0 
#time 0 0 0 0 0 0 
#perc 100 0 0 0 0 0 
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	2@0  ]
[ 2@0,	3@0  ]
[ 3@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	2@0  ]
[ 2@0,	3@0  ]
[ 3@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	2@0  ]
[ 2@0,	3@0  ]
[ 3@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	2@0  ]
[ 2@0,	3@0  ]
[ 3@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	1@0  ]
[ 1@0,	4@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	1@0  ]
[ 1@0,	4@0  ]
#--------------------------#
# t	h  # name
# void	void  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	1@0  ]
[ 2@0,	2@0  ]
[ 3@0,	3@0  ]
[ 4@0,	4@0  ]
[ 5@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	1@0  ]
[ 2@0,	2@0  ]
[ 3@0,	3@0  ]
[ 4@0,	4@0  ]
[ 5@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	void  # type
#--------------------------#
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ "candidate stuff" ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	2@0  ]
[ 1@0,	5@0  ]
#--------------------------#
# h	t  # name
# void	void  # type
#--------------------------#
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#

# 17:24:14 >  
# 17:24:14 >  "Done."
# 17:24:14 >  

